components/Slider.js

```diff
@@ -0,0 +1,262 @@
+        const clientX = evt.touches ? evt.touches[0].clientX : evt.clientX;
+        const track = this.track.getBoundingClientRect();
+        const unrounded = ((clientX - track.left) / track.width);
+        const rounded = Math.round(((range * unrounded) / step)) * step;
+        left = (((rounded - min) / range) * 100);
+        newValue = rounded;
+      }
+    }
+
+    if (newValue <= Number(min)) {
+      left = 0;
+      newValue = min;
+    }
+    if (newValue >= Number(max)) {
+      left = 100;
+      newValue = max;
+    }
+
+    return { left, newValue };
+  }
+
+  handleMouseStart = () => {
+    this.element.ownerDocument.addEventListener('mousemove', this.updatePosition)
+    this.element.ownerDocument.addEventListener('mouseup', this.handleMouseEnd)
+  }
+
+  handleMouseEnd = () => {
+    this.element.ownerDocument.removeEventListener('mousemove', this.updatePosition)
+    this.element.ownerDocument.removeEventListener('mouseup', this.handleMouseEnd)
+  }
+
+  handleTouchStart = () => {
+    this.element.ownerDocument.addEventListener('touchmove', this.updatePosition);
+    this.element.ownerDocument.addEventListener('touchup', this.handleTouchEnd);
+    this.element.ownerDocument.addEventListener('touchend', this.handleTouchEnd);
+    this.element.ownerDocument.addEventListener('touchcancel', this.handleTouchEnd);
+  }
+
+  handleTouchEnd = () => {
+    this.element.ownerDocument.removeEventListener('touchmove', this.updatePosition);
+    this.element.ownerDocument.removeEventListener('touchup', this.handleTouchEnd);
+    this.element.ownerDocument.removeEventListener('touchend', this.handleTouchEnd);
+    this.element.ownerDocument.removeEventListener('touchcancel', this.handleTouchEnd);
+  }
+
+  handleChange = (evt) => {
+    this.setState({ value: evt.target.value });
+    this.updatePosition();
+  }
+
+  render() {
+    const {
+      className,
+      hideTextInput,
+      id,
+      onChange,
+      min,
+      minLabel,
+      max,
+      maxLabel,
+      labelText, 
+      step,
+      required,
+      disabled,
+      ...other
+    } = this.props;
+
+    const {
+      value,
+      left,
+    } = this.state;
+
+    const sliderClasses = classNames(
+      'bx--slider',
+      { 'bx--slider--disabled': disabled },
+      className,
+    );
+
+    const filledTrackStyle = {
+      transform: `translate(0%, -50%) scaleX(${left / 100})`,
+    }
+    const thumbStyle = {
+      left: `${left}%`,
+    }
+
+    return (
+      <div className="bx--form-item">
+        <label htmlFor={id} className="bx--label">{labelText}</label>
+        <div className="bx--slider-container">
+          <span className="bx--slider__range-label">{min}{minLabel}</span>
+          <div
+            className={sliderClasses}
+            ref={node => {
+              this.element = node;
+            }}
+            onClick={(evt) => this.updatePosition(evt)}
+            {...other}
```

Comment:
I think the `{ ...other }` props should be on the `<input>`, at least for me, I need to add `name` attribute in order for this component to submit data. And any other handlers as well

[![Edit Form Inputs with submit](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/DRr8zoDK6)

Maybe instead, we provide a `name` prop that the user can use and `...other` can go